query/worker.go

```diff
@@ -133,7 +131,22 @@ func (w *worker) Run(results chan<- *jobResult, quit <-chan struct{}) {
 			log.Tracef("Worker %v queuing job %T with index %v",
 				peer.Addr(), job.Req, job.Index())
 
-			peer.QueueMessageWithEncoding(job.Req, nil, job.encoding)
+			err := job.SendQuery(peer, job.Req)
+
+			// If any error occurs while sending query, quickly send the result
+			// containing the error to the workmanager.
+			if err != nil {
+				select {
+				case results <- &jobResult{
+					job:  job,
+					peer: peer,
+					err:  err,
+				}:
+				case <-quit:
+					return
+				}
+				goto nexJobLoop
```

Comment:
Would a `continue` suffice here? Is there anything that this `goto` provides that a `continue` wouldn't do?

Reply:
Thanks looking at it again, I think `continue` could work, thanks.